tls/s2n_prf.c

```diff
@@ -445,6 +446,36 @@ int s2n_hybrid_prf_master_secret(struct s2n_connection *conn, struct s2n_blob *p
     return s2n_prf(conn, premaster_secret, &label, &client_random, &server_random, &conn->kex_params.client_key_exchange_message, &master_secret);
 }
 
+int s2n_prf_calculate_master_secret(struct s2n_connection *conn, struct s2n_blob *premaster_secret)
+{
+    POSIX_ENSURE_REF(conn);
+
+    POSIX_ENSURE_EQ(s2n_conn_get_current_message_type(conn), CLIENT_KEY);
+
+    /* TODO: https://github.com/aws/s2n-tls/issues/2990 */
+    if (conn->ems_negotiated && S2N_IN_TEST) {
+        /* Only the client writes the Client Key Exchange message */
+        if (conn->mode == S2N_CLIENT) {
+            POSIX_GUARD(s2n_handshake_finish_header(&conn->handshake.io));
+        }
+        struct s2n_stuffer client_key_message = conn->handshake.io;
+        POSIX_GUARD(s2n_stuffer_reread(&client_key_message));
+        uint8_t client_key_message_size = s2n_stuffer_data_available(&client_key_message);
+        struct s2n_blob client_key_blob = { 0 };
+        POSIX_ENSURE_REF(client_key_blob.data = s2n_stuffer_raw_read(&client_key_message, client_key_message_size));
+        client_key_blob.size = client_key_message_size;
```

Comment:
This is going to be tricky to make clear. I'm guessing you're not just using conn->handshake.io.blob directly because manipulating the sized caused problems?

I'd suggest either the short but non-API way:
```suggestion
        struct s2n_blob client_key_blob = { 0 };
        POSIX_GUARD(s2n_blob_init(&client_key_blob, client_key_message.blob.data, client_key_message_size));
```

Or the full API way:
```suggestion
        struct s2n_blob client_key_blob = { 0 };
        uint8_t client_key_size = s2n_stuffer_data_available(&client_key_message);
        uint8_t client_key_data = s2n_stuffer_raw_read(&client_key_message, client_key_message_size)
        POSIX_GUARD(s2n_blob_init(&client_key_blob, client_key_data, client_key_size));
```

I think mixing the two methods loses readability.